kustomize/commands/build/build_benchmark_test.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2023 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package build_test
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	. "sigs.k8s.io/kustomize/kustomize/v5/commands/build"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+type GenConfig struct {
+	fileResources     int
+	resources         int
+	patches           int
+	namespaced        bool
+	namePrefix        string
+	nameSuffix        string
+	commonLabels      map[string]string
+	commonAnnotations map[string]string
+}
+
+var genConfig = []GenConfig{
+	{
+		resources:  4,
+		namePrefix: "foo-",
+		nameSuffix: "-bar",
+		commonLabels: map[string]string{
+			"foo": "bar",
+		},
+		commonAnnotations: map[string]string{
+			"baz": "blatti",
+		},
+	},
+	{
+		resources: 100,
+	},
+	{
+		resources: 3,
+	},
+	{
+		resources:     2,
+		namespaced:    true,
+		fileResources: 30,
+		patches:       10,
+	},
+	{
+		fileResources: 2,
+	},
+}
+
+func makeKustomization(fSys filesys.FileSystem, path, id string, depth int) error {
+	cfg := genConfig[depth]
+	fSys.MkdirAll(path)
+
+	var buf bytes.Buffer
+	if cfg.namespaced {
+		fmt.Fprintf(&buf, "namespace: %s\n", id)
+	}
+
+	if cfg.namePrefix != "" {
+		fmt.Fprintf(&buf, "namePrefix: %s\n", cfg.namePrefix)
+	}
+
+	if cfg.nameSuffix != "" {
+		fmt.Fprintf(&buf, "nameSuffix: %s\n", cfg.nameSuffix)
+	}
+
+	if len(cfg.commonLabels) > 0 {
+		fmt.Fprintf(&buf, "commonLabels:\n")
+		for k, v := range cfg.commonLabels {
+			fmt.Fprintf(&buf, "  %s: %s\n", k, v)
+		}
+	}
+
+	if len(cfg.commonAnnotations) > 0 {
+		fmt.Fprintf(&buf, "commonAnnotations:\n")
+		for k, v := range cfg.commonAnnotations {
+			fmt.Fprintf(&buf, "  %s: %s\n", k, v)
+		}
+	}
+
+	if cfg.fileResources > 0 || cfg.resources > 0 {
+		fmt.Fprintf(&buf, "resources:\n")
+		for res := 0; res < cfg.fileResources; res++ {
+			fn := fmt.Sprintf("res%d.yaml", res)
+			fmt.Fprintf(&buf, " - %v\n", fn)
+
+			buf := fmt.Sprintf(`kind: ConfigMap
```

Comment:
Good point, that was confusing. to avoid too long lines when fixing `Error return value of `fSys.WriteFile` is not checked (errcheck)` I think it should be a variable